llmfoundry/data/denoising.py

```diff
@@ -678,7 +677,8 @@ def _sample_span_lengths(total_tokens: int, num_spans: int) -> np.ndarray:
         """
         span_markers = np.less(np.arange(total_tokens - 1), num_spans -
                                1)[np.random.permutation(total_tokens - 1)]
-        span_start_indicator = np.concatenate([[0], span_markers])
+        span_start_indicator = np.concatenate([[0],
+                                               span_markers])  # type: ignore
```

Comment:
This seems to be pure numpy types... why do we need type ignore here? Same with all of the below.

Reply:
Yeah, I do not know. I accepted these type ignores in favor of attempting to actually get this pr in. There are a variety of type ignores throughout the repo that should be revisited, but I figured better to stop making things worse. Trade off of how much time I spend on this pr vs reward.